LibOS/shim/src/bookkeep/shim_signal.c

```diff
@@ -295,6 +304,71 @@ static void memfault_upcall (PAL_PTR event, PAL_NUM arg, PAL_CONTEXT * context)
     DkExceptionReturn(event);
 }
 
+bool test_user_memory (void * addr, size_t size, bool write)
```

Comment:
Same request - add a comment documenting the function's job and expected behavior.